routingpy/routers/ign.py

```diff
@@ -0,0 +1,409 @@
+        Initializes an IGN client.
+
+        :param base_url: The base URL for the request. Defaults to the official address
+            instance for "car". Should not have a trailing slash.
+
+        :param user_agent: User Agent to be used when requesting.
+            Default :attr:`routingpy.routers.options.default_user_agent`.
+
+        :param timeout: Combined connect and read timeout for HTTP requests, in
+            seconds. Specify ``None`` for no timeout. Default :attr:`routingpy.routers.options.default_timeout`.
+
+        :param retry_timeout: Timeout across multiple retriable requests, in
+            seconds.  Default :attr:`routingpy.routers.options.default_retry_timeout`.
+
+        :param retry_over_query_limit: If True, client will not raise an exception
+            on HTTP 429, but instead jitter a sleeping timer to pause between
+            requests until HTTP 200 or retry_timeout is reached.
+            Default :attr:`routingpy.routers.options.default_retry_over_query_limit`.
+
+        :param skip_api_error: Continue with batch processing if a :class:`routingpy.exceptions.RouterApiError` is
+            encountered (e.g. no route found). If False, processing will discontinue and raise an error.
+            Default :attr:`routingpy.routers.options.default_skip_api_error`.
+
+        :param client: A client class for request handling. Needs to be derived from :class:`routingpy.client_base.BaseClient`
+
+        :param client_kwargs: Additional arguments passed to the client, such as headers or proxies.
+        """
+
+        self.client = client(
+            base_url,
+            user_agent,
+            timeout,
+            retry_timeout,
+            retry_over_query_limit,
+            skip_api_error,
+            **client_kwargs,
+        )
+
+    def directions(
+        self,
+        locations: List[List[float]] = [],
+        profile: Optional[str] = "car",
+        resource: str = "bdtopo-osrm",
+        optimization: Optional[str] = None,
+        geometry_format: Optional[str] = None,
+        constraints: Optional[Union[str, dict, List[str]]] = None,
+        getSteps: Optional[bool] = None,
+        getBbox: Optional[bool] = None,
+        crs: Optional[str] = None,
+        waysAttributes: Optional[List[str]] = None,
+        dry_run: Optional[bool] = None,
+        **direction_kwargs,
+    ):
+        """
+        Get directions between an origin point and a destination point.
+
+        For more information, visit https://www.geoportail.gouv.fr/depot/swagger/itineraire.html
+
+        Use ``direction_kwargs`` for any missing ``directions`` request options.
+
+        :param locations: The coordinates tuple the route should be calculated
+            from in order of visit.
+
+        :param profile: Optionally specifies the mode of transport to use when calculating
+            directions. Should be "car" (default) or "pedestrian";
+            resource "bdtopo-osrm" also supports "exceptionnal".
```

Comment:
1. I think there is typo, and it should be "exceptional"
2. I'm not entirely sure what you mean by that. Could you elaborate? (At least in your reply here. Perhaps it's sufficient in the documentation and it's just me who doesn't understand what it's about.)

Reply:
My bad, I wrote this comment too quickly. I see this “exceptionnal” in the response from getCapabilites, and the typo is on IGN's side!